packages/fast-jss-manager-angular/src/manage-jss.ts

```diff
@@ -0,0 +1,295 @@
+
+                if (Boolean(dynamicStyles)) {
+                    state.dynamicStyleSheet = jss.createStyleSheet(
+                        JSSManager.separatedStyles[this.instanceId].dynamicStyles,
+                        { link: true }
+                    );
+                }
+
+                this.state = state;
+
+                this.className = this.getClassNames()[(Object.keys(styles)[0] as any)];
+            }
+
+            private ngAfterViewInit(): void {
+                if (super.ngAfterViewInit) {
+                    super.ngAfterViewInit();
+                }
+
+                if (Boolean(this.state.staticStyleSheet)) {
+                    JSSManager.stylesheetManager.add(this.state.staticStyleSheet, this.state.staticStyleSheet);
+                    JSSManager.stylesheetManager.manage(this.state.staticStyleSheet);
+                }
+
+                if (Boolean(this.state.dynamicStyleSheet)) {
+                    // It appears we need to update the stylesheet for any style properties defined as functions
+                    // to work.
+                    // TODO: We"ll need to call this with a context if it exists
+                    this.state.dynamicStyleSheet.attach().update(this.designSystem);
+                }
+            }
+
+            private ngOnDestroy(): void {
+                if (this.hasStaticStyleSheet()) {
+                    JSSManager.stylesheetManager.unmanage(this.state.staticStyleSheet);
+                }
+
+                if (this.hasDynamicStyleSheet()) {
+                    this.state.dynamicStyleSheet.detach();
+                    jss.removeStyleSheet(this.state.dynamicStyleSheet);
+                }
+
+                const deregisterComponentEvent: CustomEvent = new CustomEvent(eventTypes.deregisterComponent);
+                this.el.nativeElement.dispatchEvent(deregisterComponentEvent);
+            }
+
+            private get designSystem(): any {
+                return this.config ? this.config : {};
+            }
+
+            /**
+             * Checks to see if this component has an associated static stylesheet
+             */
+            private hasStaticStyleSheet(): boolean {
+                return Boolean(this.state.staticStyleSheet);
+            }
+
+            /**
+             * Checks to see if this component has an associated dynamic stylesheet
+             */
+            private hasDynamicStyleSheet(): boolean {
+                return Boolean(this.state.dynamicStyleSheet);
+            }
+
+            /**
+             * Separates a single ComponentStyles object into an ISeparatedStylesheet object
+             * If either a dynamic or static stylesheet is empty (there are no styles) then that
+             * key will not be created.
+             */
+            private separateStyles(componentStyles: ComponentStyles<S, C>): ISeparatedStylesheet<S, C> {
+                // TODO: write a test for this method to make sure it always returns an object.
```

Comment:
For every TODO statement in this PR create issues and update the TODO statement with number and title.